### PR TITLE
chore: [LWG-958]: Refactored DNS link code to reduce complexity

### PR DESCRIPTION
**Pull request by rishik30**
##### Summary:

DNSLink component held complex logic and all components in a single functional component which made it very hard to maintain and edit. Also, the sonar complexity score was more than 300.
So, refactored the whole code and reduced the complexity below 25.

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/modules/75-ce/components/COGatewayAccess/LBAdvancedConfiguration.tsx [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+      <Tabs
+        id="tabsId1"
+        selectedTabId={activeConfigTabId}
+        onChange={tabId => tabId !== activeConfigTabId && setActiveConfigTabId(tabId as string)}
+      >
+        <Tab
+          id="routing"
+          title="Routing"
+          panel={
+            <Container style={{ backgroundColor: '#FBFBFB' }}>
```
Comment (jenil-harness) [inline src/modules/75-ce/components/COGatewayAccess/LBAdvancedConfiguration.tsx:148]: `Color`

File: src/modules/75-ce/components/COGatewayAccess/LBAdvancedConfiguration.tsx [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+  return (
+    <Container className={css.dnsLinkContainer}>
+      <Tabs
+        id="tabsId1"
+        selectedTabId={activeConfigTabId}
+        onChange={tabId => tabId !== activeConfigTabId && setActiveConfigTabId(tabId as string)}
+      >
+        <Tab
+          id="routing"
+          title="Routing"
```
Comment (jenil-harness) [inline src/modules/75-ce/components/COGatewayAccess/LBAdvancedConfiguration.tsx:146]: To strings

File: src/modules/75-ce/components/COGatewayAccess/LBAdvancedConfiguration.tsx [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+                <>
+                  <Text className={css.titleHelpTextDescription}>
+                    {getString('ce.co.gatewayConfig.routingDescription')}
+                  </Text>
+                  <Layout.Vertical spacing="large">
+                    {!showRoutingTable || loadingSecurityGroups ? (
+                      <Icon
+                        name="spinner"
+                        size={24}
+                        color="blue500"
```
Comment (jenil-harness) [inline src/modules/75-ce/components/COGatewayAccess/LBAdvancedConfiguration.tsx:159]: `Color`

File: src/modules/75-ce/components/COGatewayAccess/LBAdvancedConfiguration.tsx [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+                </>
+              )}
+            </Container>
+          }
+        />
+        <Tab
+          id="healthcheck"
+          title="Health check"
+          panel={
+            <Container style={{ backgroundColor: '#FBFBFB' }}>
```
Comment (jenil-harness) [inline src/modules/75-ce/components/COGatewayAccess/LBAdvancedConfiguration.tsx:175]: `Color` & string.yml for title

File: src/modules/75-ce/components/COGatewayAccess/LoadBalancerSelection.tsx [hunk 5]
```diff
@@ -0,0 +1,360 @@
...
+  const [selectedLoadBalancer, setSelectedLoadBalancer] = useState<AccessPointCore>()
+  const [accessPoint, setAccessPoint] = useState<AccessPoint>()
+  const [isCreateMode, setIsCreateMode] = useState<boolean>(false)
+
+  const {
+    data: accessPoints,
+    loading: accessPointsLoading,
+    refetch: refetchAccessPoints
+  } = useListAccessPoints({
+    account_id: accountId, // eslint-disable-line
```
Comment (jenil-harness) [inline src/modules/75-ce/components/COGatewayAccess/LoadBalancerSelection.tsx:153]: eslint disable?

File: src/modules/75-ce/components/COGatewayAccess/ResourceAccessUrlSelector.tsx [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+        })
+    }, 500),
+    [gatewayDetails]
+  )
+
+  return (
+    <>
+      <Container className={css.dnsLinkContainer}>
+        <Layout.Horizontal spacing="small" style={{ marginBottom: 'var(--spacing-xlarge)' }}>
+          <Heading level={3} font={{ weight: 'light' }}>
```
Comment (jenil-harness) [inline src/modules/75-ce/components/COGatewayAccess/ResourceAccessUrlSelector.tsx:128]: use `fontVariation`

File: src/modules/75-ce/components/COGatewayAccess/ResourceAccessUrlSelector.tsx [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+            data-testid="noCustomDomain"
+          />
+          <Layout.Vertical spacing="xsmall">
+            <Text
+              color={Color.GREY_500}
+              style={{
+                fontSize: '12px',
+                fontWeight: 400,
+                lineHeight: '18px',
+                color: 'var(--grey-800)',
```
Comment (jenil-harness) [inline src/modules/75-ce/components/COGatewayAccess/ResourceAccessUrlSelector.tsx:154]: Two different colors?

File: src/modules/75-ce/components/COGatewayAccess/helper.ts [hunk 8]
```diff
@@ -100,6 +121,69 @@ export const getDummyGcpSupportedResourceFromLb = (lb: AccessPoint): AccessPoint
...
+  accountId,
+  projectId,
+  orgId
+}: GetInitialAccessPointDetails): AccessPoint => ({
+  cloud_account_id: gatewayDetails.cloudAccount.id, // eslint-disable-line
+  account_id: accountId, // eslint-disable-line
+  project_id: projectId, // eslint-disable-line
+  org_id: orgId, // eslint-disable-line
+  metadata: {
+    security_groups: [], // eslint-disable-line
```
Comment (jenil-harness) [inline src/modules/75-ce/components/COGatewayAccess/helper.ts:143]: eslint disable?

**Review by jenil-harness — CHANGES_REQUESTED** [review]

File: src/modules/75-ce/components/COGatewayAccess/CustomDomainMapping.tsx [hunk 9]
```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2021 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React from 'react'
+import { Container, FormInput, Heading, Layout, SelectOption, Text } from '@wings-software/uicore'
+import { Radio } from '@blueprintjs/core'
```
[the remark this answers is not in the document]
Comment (rishik30) [reply src/modules/75-ce/components/COGatewayAccess/CustomDomainMapping.tsx:10]: will do in a follow-up PR as it requires style changes as well.

File: src/modules/75-ce/components/COGatewayAccess/LBAdvancedConfiguration.tsx (hunk 1 above)
Comment (rishik30) [reply src/modules/75-ce/components/COGatewayAccess/LBAdvancedConfiguration.tsx:148]: will take it in the follow-up PR.

**Review by jenil-harness — APPROVED** [review]

**Review by rajarshee-harness — APPROVED** [review]

**Pull request merged**